### PR TITLE
Automate `stable_abi.txt` updates

### LINKED ISSUE #5: Automate `stable_abi.txt` updates
We can create a cron workflow on GitHub Actions that updates `stable_abi.txt` via pull request automatically.

Resources:
1. https://github.com/marketplace/actions/create-pull-request

### PR DESCRIPTION
**Pull request by messense**
Closes #5

### CODE REVIEW

**Comment by adamreichold** [comment]
Will this open a PR even if there are no changes to the file?

File: .github/workflows/update.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        curl https://raw.githubusercontent.com/python/cpython/main/Misc/stable_abi.txt > Misc/stable_abi.txt
+    - name: Create Pull Request
+      uses: peter-evans/create-pull-request@v4
+      with:
+        delete-branch: true
+        add-paths: |
+          Misc/stable_abi.txt
+        title: 'Update stable_abi.txt to the latest main'
+        commit-message: 'chore: Update stable_abi.txt to the latest main'
+        body: 'Source: https://raw.githubusercontent.com/python/cpython/main/Misc/stable_abi.txt'
```
Comment (ravenexp) [inline .github/workflows/update.yml:26]: What happens if the commit aborts because there are no changes?

**Comment by messense** [comment]
https://github.com/marketplace/actions/create-pull-request#action-behaviour

> Will this open a PR even if there are no changes to the file?

> What happens if the commit aborts because there are no changes?

 If there are no changes (i.e. no diff exists with the checked-out base branch), no pull request will be created and the action exits silently.

**Comment by messense** [comment]
I'll merge it and trigger it manually to verify it works.

**Comment by messense** [comment]
First working PR: https://github.com/PyO3/python3-dll-a/pull/7

**Comment by messense** [comment]
Second run with no changes to `stable_abi.txt` does not create new PR and exits silently: https://github.com/PyO3/python3-dll-a/runs/6071326864?check_suite_focus=true

So it works!

**Comment by ravenexp** [comment]
It works great! Thank you!

**Pull request merged**